src/statuscode.cpp

```diff
@@ -6,16 +6,23 @@
  */
 
 #include <statuscode.h>
-#include <Utils.h>
+#include <sstream>
+
+template<typename T>
+static std::string toHexString (const T t)
+{
+        std::ostringstream oss;
+        oss << std::hex << (unsigned long)t << std::dec;
```

Comment:
Is c-style casting of any type T to unsigned long dangerous? What if I call this function with some instance of a complex type, how would the function react (undefined?).

Reply:
Hello, in my opinion it is safe because:
1) it's not a pointer and it doesn't involve 'wild' point dereference,
2) if a complex C++ type ( e.g. a class ) doesn't implement   long operator()  then it will simple not compile.

However If that's not convincing enough, please propose a better solution.